onboarding/views.py

```diff
@@ -55,16 +55,20 @@ def track_topic_view(request, trackid):
 def topic_course_view(request, track_name, topicid):
     
     courses = TopicCourseBridge.objects.filter(topic=topicid).values_list('course_id', 'course__name', 'course__descirption', 'course__image', 'topic__name', 'topic__id')
-    
-    return render(request, 'topic_courses.html', {'courses': courses, 'track_name':track_name, 'topic_id':topicid})
+    track = Track.objects.filter(name=track_name).values_list('id')
```

Comment:
Maybe you should consider doing:
Track.objects.get(name=track_name).id

Reply:
OR change the def topic_course_view(request, track_name, topicid):
To def topic_course_view(request, track_id, topicid), then you can do:
track = Track.objects.get(id=track_track_id)
track_name = track.name
track_id = track.id
Always try to make the queryset using ID since it's unique